src/Interpreters/Cache/FileCache.cpp

```diff
@@ -1400,6 +1415,7 @@ FileCache::~FileCache()
 void FileCache::deactivateBackgroundOperations()
 {
     shutdown.store(true);
+    stop_loading_metadata = true;
```

Comment:
Ok, but also we need to do `join` on the loading thread. So better to remove `load_metadata_main_thread.detach();` and save the thread inside the class.